packages/client/src/forms/configuration/administrative/addresses.ts

```diff
@@ -768,7 +768,7 @@ function getAddressCaseFields(
         },
         {
           action: 'hide',
-          expression: 'values.ruralOrUrbanPrimary !== "URBAN"'
+          expression: `values.ruralOrUrban${sentenceCase(useCase)} !== "URBAN"`
```

Comment:
Would it be possible to do something like this instead:
```
(values.locationType === 'primary' && values.ruralOrUrbanPrimary !== 'URBAN') ||
(values.locationType === 'secondary' && values.ruralOrUrbanSecondary !== 'URBAN')
```

I'm not 100% sure in which field the location type (primary/secondary) is, but I assume you get the point.  It's probably best if we get rid of these non-typed string expressions at some point so it would be great if we'd keep them as simple as possible.